sklearn/neighbors/_base.py

```diff
@@ -1021,6 +1021,16 @@ class from an array representing our data set and ask who's
                 neigh_ind_list = sum(chunked_results, [])
                 results = _to_object_array(neigh_ind_list)
 
+            if sort_results:
+                if not return_distance:
+                    raise ValueError("return_distance must be True "
+                                     "if sort_results is True.")
+                for ii in range(len(neigh_dist)):
```

Comment:
Is there a way we can vectorize this operation instead of looping over all samples? IIUC the arrays in `neigh_dist` may be of different lengths so that might not be possible, but I'm not familiar enough to confirm

Reply:
It is not possible, as you said each row is an array of different length.